### PR TITLE
SNOW-1794920: Test more Series.map cases and refactor Series apply/map tests.

### PR DESCRIPTION
**Pull request by sfc-gh-mvashishtha**
Fixes SNOW-1794920

- All the test cases that test a callable mapper with no other arguments now test both `apply` and `map` and belong to a single test class.
- All the test cases that should test `apply` only are now in a separate test class.
- All the test cases that should test `map` only are now in a separate test class.

### CODE REVIEW

File: tests/integ/modin/series/test_apply_and_map.py [hunk 1]
```diff
@@ -0,0 +1,728 @@
...
+    param(
+        [native_pd.Timedelta(1), native_pd.Timedelta(2)],
+        lambda x: x.value,
+        "int",
+        id="apply_on_frame_with_timedelta_data_columns_returns_int",
+        marks=pytest.mark.xfail(strict=True, raises=NotImplementedError),
+    ),
+]
+
+# TODO SNOW-876999: Test return date/time/timestamp type when
```
Comment (sfc-gh-helmeleegy) [inline tests/integ/modin/series/test_apply_and_map.py:67]: This is marked as done. Can we either remove the todo (if truly resolved) or create a new ticket to capture the remaining work to be done?

**Review by sfc-gh-helmeleegy — COMMENTED** [review]
Looks good overall. Just one request to fix an outdated todo.

File: tests/integ/modin/series/test_apply_and_map.py (hunk 1 above)
Comment (sfc-gh-mvashishtha) [reply tests/integ/modin/series/test_apply_and_map.py]: There's follow-up work required. I filed a new issue, SNOW-1798212, and I linked to it here.

**Review by sfc-gh-jkew — APPROVED** [review]
LGTM

**Review by sfc-gh-helmeleegy — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**